repos/system_upgrade/el7toel8/workflows/inplace_upgrade.py

```diff
@@ -16,118 +16,158 @@ class IPUWorkflow(Workflow):
         flags = Flags()
 
     class DownloadPhase(Phase):
+        '''Download of all things needed (RPMs, Kernel images, Anaconda image
+        (very big maybe), etc.)
+        '''
         name = 'Download'
         filter = TagFilter(DownloadPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class InterimPreparationPhase(Phase):
+        '''Preparation of initial ramdisk (if required).
+        Setup bootloaderGrub.
+        '''
         name = 'InterimPreparation'
         filter = TagFilter(InterimPreparationPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags(request_restart_after_phase=True)
 
     class InitRamStartPhase(Phase):
+        '''Reboot into the next system therefore not a real step, however the
+        before and post parts are interesting to have for hooking.
+        '''
         name = 'InitRamStart'
         filter = TagFilter(InitRamStartPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class NetworkPhase(Phase):
+        '''Bring up the network.'''
         name = 'Network'
         filter = TagFilter(NetworkPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class StoragePhase(Phase):
+        '''Mount all storage points.'''
         name = 'Storage'
         filter = TagFilter(StoragePhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class LateTestsPhase(Phase):
+        '''Tests that need to be executed later.'''
         name = 'LateTests'
         filter = TagFilter(LateTestsPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class PreparationPhase(Phase):
+        '''YUM Configurations. Backup not upgradable configurations (No Verify
+        Configs).
+        '''
         name = 'Preparation'
         filter = TagFilter(PreparationPhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class RPMUpgradePhase(Phase):
+        '''Upgrade RPMs. Check all files are installed (troubles with handling
+        dirs/symlinks), includes reinstallation of affected packages.
+        '''
         name = 'RPMUpgrade'
         filter = TagFilter(RPMUpgradePhaseTag)
         policies = Policies(Policies.Errors.FailPhase,
                             Policies.Retry.Phase)
         flags = Flags()
 
     class ApplicationsPhase(Phase):
+        '''Application of changes on configuration files, move various
+        directores, etc.
+        '''
```

Comment:
Provide all neccessary steps to finish upgrade of particular native applications. Including move/rename of configuration files (if required), modify configuration of applications to be able to run correctly (if possible, to be set up as close as possible to original behaviour), ....